<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

Let us now load the ipython-sql  extension and establish a connection with the database

*   Here you will be loading the csv files into the pandas Dataframe and then loading the data into the above mentioned sqlite tables.

*   Next you will be connecting to the sqlite database  **FinalDB**.

Refer to the previous lab for hints .

<a href ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing_SQLite.ipynb?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Hands-on Lab: Analyzing a real World Data Set</a>


In [1]:
%load_ext sql

ModuleNotFoundError: No module named 'sql'

In [2]:
import pandas as pd
import sqlite3

In [3]:
!pip install ipython-sql

In [4]:
df1=pd.read_csv('Chicago_Crime_Data-v2.csv')
df2=pd.DataFrame(df1)
df2



ID CASE_NUMBER                    DATE                     BLOCK  \
0     3512276    HK587712  08/28/2004 05:50:56 PM        047XX S KEDZIE AVE   
1     3406613    HK456306  06/26/2004 12:40:00 PM  009XX N CENTRAL PARK AVE   
2     8002131    HT233595  04/04/2011 05:45:00 AM        043XX S WABASH AVE   
3     7903289    HT133522  12/30/2010 04:30:00 PM      083XX S KINGSTON AVE   
4    10402076    HZ138551  02/02/2016 07:30:00 PM           033XX W 66TH ST   
..        ...         ...                     ...                       ...   
528  10453948    HZ192829  03/01/2016 12:00:00 AM           028XX E 79TH ST   
529  10397129    HZ133234  01/29/2016 03:30:00 PM       006XX W HARRISON ST   
530   3269495    HJ747227  11/08/2003 04:54:15 PM           012XX W 81ST ST   
531  10840565    JA143710  02/06/2017 01:20:00 PM        009XX N KARLOV AVE   
532   1326195     G021609  01/11/2001 02:30:41 AM       087XX S ESCANABA AV   

     IUCR                      PRIMARY_TYPE  \
0     890                             THEFT   
1     820                             THEFT   
2     820                             THEFT   
3     840                             THEFT   
4     820                             THEFT   
..    ...                               ...   
528  1055                 HUMAN TRAFFICKING   
529  5114                    NON - CRIMINAL   
530   510                         RITUALISM   
531  5073  NON-CRIMINAL (SUBJECT SPECIFIED)   
532  9901                 DOMESTIC VIOLENCE   

                                     DESCRIPTION  \
0                                  FROM BUILDING   
1                                 $500 AND UNDER   
2                                 $500 AND UNDER   
3                  FINANCIAL ID THEFT: OVER $300   
4                                 $500 AND UNDER   
..                                           ...   
528                        INVOLUNTARY SERVITUDE   
529                            FOID - REVOCATION   
530  AGG RIT MUT: HANDS/FIST/FEET SERIOUS INJURY   
531       NOTIFICATION OF CIVIL NO CONTACT ORDER   
532                            DOMESTIC VIOLENCE   

             LOCATION_DESCRIPTION  ARREST  DOMESTIC  ...  WARD  \
0              SMALL RETAIL STORE   False     False  ...  14.0   
1                           OTHER   False     False  ...  27.0   
2    NURSING HOME/RETIREMENT HOME   False     False  ...   3.0   
3                       RESIDENCE   False     False  ...   7.0   
4                           ALLEY   False     False  ...  15.0   
..                            ...     ...       ...  ...   ...   
528                     APARTMENT   False     False  ...   7.0   
529                        STREET   False     False  ...   2.0   
530                         OTHER   False     False  ...  21.0   
531                     RESIDENCE   False      True  ...  37.0   
532                     APARTMENT    True      True  ...   NaN   

     COMMUNITY_AREA_NUMBER  FBICODE  X_COORDINATE Y_COORDINATE  YEAR  \
0                     58.0        6     1155838.0    1873050.0  2004   
1                     23.0        6     1152206.0    1906127.0  2004   
2                     38.0        6     1177436.0    1876313.0  2011   
3                     46.0        6     1194622.0    1850125.0  2010   
4                     66.0        6     1155240.0    1860661.0  2016   
..                     ...      ...           ...          ...   ...   
528                   46.0       26     1196679.0    1853139.0  2016   
529                   28.0       26     1172257.0    1897564.0  2016   
530                   71.0      04B     1169648.0    1851076.0  2003   
531                   23.0       26     1148881.0    1905963.0  2017   
532                    NaN      08B     1196869.0    1847416.0  2001   

                  UPDATEDON   LATITUDE  LONGITUDE  \
0    02/10/2018 03:50:01 PM  41.807440 -87.703956   
1    02/28/2018 03:56:25 PM  41.898280 -87.716406   
2    02/10/2018 03:50:01 PM  41.815933 -87.624642   
3 

In [33]:
df3=pd.read_csv('Census_Data.csv')
df3=pd.DataFrame(df3)
df3

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT OF HOUSING CROWDED  \
0                     1.0         Rogers Park                         7.7   
1                     2.0          West Ridge                         7.8   
2                     3.0              Uptown                         3.8   
3                     4.0      Lincoln Square                         3.4   
4                     5.0        North Center                         0.3   
..                    ...                 ...                         ...   
73                   74.0     Mount Greenwood                         1.0   
74                   75.0         Morgan Park                         0.8   
75                   76.0              O'Hare                         3.6   
76                   77.0           Edgewater                         4.1   
77                    NaN             CHICAGO                         4.7   

    PERCENT HOUSEHOLDS BELOW POVERTY  PERCENT AGED 16+ UNEMPLOYED  \
0                               23.6                          8.7   
1                               17.2                          8.8   
2                               24.0                          8.9   
3                               10.9                          8.2   
4                                7.5                          5.2   
..                               ...                          ...   
73                               3.4                          8.7   
74                              13.2                         15.0   
75                              15.4                          7.1   
76                              18.2                          9.2   
77                              19.7                         12.9   

    PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  \
0                                           18.2   
1                                           20.8   
2                                           11.8   
3                                           13.4   
4                                            4.5   
..                                           ...   
73                                           4.3   
74                                          10.8   
75                                          10.9   
76                                           9.7   
77                                          19.5   

    PERCENT AGED UNDER 18 OR OVER 64  PER_CAPITA_INCOME   HARDSHIP_INDEX  
0                               27.5               23939            39.0  
1                               38.5               23040            46.0  
2                               22.2               35787            20.0  
3                               25.5               37524            17.0  
4                               26.2               57123             6.0  
..                               ...                 ...             ...  
73                              36.8               34381            16.0  
74                              40.3               27149            30.0  
75                              30.3               25828            24.0  
76                              23.8               33385            19.0  
77                              33.5               28202             NaN  

[78 rows x 9 columns]

In [12]:
df4=pd.read_csv('Chicago_Public_Schools.csv')
df4=pd.DataFrame(df4)
df4

School ID                                     NAME_OF_SCHOOL  \
0       610038                  Abraham Lincoln Elementary School   
1       610281  Adam Clayton Powell Paideia Community Academy ...   
2       610185                Adlai E Stevenson Elementary School   
3       609993                    Agustin Lara Elementary Academy   
4       610513                      Air Force Academy High School   
..         ...                                                ...   
561     610172                William T Sherman Elementary School   
562     609844                 William W Carter Elementary School   
563     610088                Wolfgang A Mozart Elementary School   
564     609977               Woodlawn Community Elementary School   
565     610392                 World Language Academy High School   

    Elementary, Middle, or High School          Street Address     City State  \
0                                   ES        615 W Kemper Pl   Chicago    IL   
1                                   ES  7511 S South Shore Dr   Chicago    IL   
2                                   ES     8010 S Kostner Ave   Chicago    IL   
3                                   ES     4619 S Wolcott Ave   Chicago    IL   
4                                   HS        3630 S Wells St   Chicago    IL   
..                                 ...                     ...      ...   ...   
561                                 ES         1000 W 52nd St   Chicago    IL   
562                                 ES    5740 S Michigan Ave   Chicago    IL   
563                                 ES      2200 N Hamlin Ave   Chicago    IL   
564                                 ES     6657 S Kimbark Ave   Chicago    IL   
565                                 HS     3120 S Kostner Ave   Chicago    IL   

     ZIP Code    Phone Number  \
0       60614  (773) 534-5720   
1       60649  (773) 535-6650   
2       60652  (773) 535-2280   
3       60609  (773) 535-4389   
4       60609  (773) 535-1590   
..        ...             ...   
561     60609  (773) 535-1757   
562     60637  (773) 535-0860   
563     60647  (773) 534-4160   
564     60637  (773) 535-0801   
565     60623  (773) 535-4334   

                                                 Link   \
0    http://schoolreports.cps.edu/SchoolProgressRep...   
1    http://schoolreports.cps.edu/SchoolProgressRep...   
2    http://schoolreports.cps.edu/SchoolProgressRep...   
3    http://schoolreports.cps.edu/SchoolProgressRep...   
4    http://schoolreports.cps.edu/SchoolProgressRep...   
..                                                 ...   
561  http://schoolreports.cps.edu/SchoolProgressRep...   
562  http://schoolreports.cps.edu/SchoolProgressRep...   
563  http://schoolreports.cps.edu/SchoolProgressRep...   
564  http://schoolreports.cps.edu/SchoolProgressRep...   
565  http://schoolreports.cps.edu/SchoolProgressRep...   

                        Network Manager  ... Freshman on Track Rate %  \
0          Fullerton Elementary Network  ...                      NDA   
1             Skyway Elementary Network  ...                      NDA   
2             Midway Elementary Network  ...                      NDA   
3           Pershing Elementary Network  ...                      NDA   
4    Southwest Side High School Network  ...                     91.8   
..                                  ...  ...                      ...   
561                        AUSL Schools  ...                      NDA   
562     Burnham Park Elementary Network  ...                      NDA   
563        Fullerton Elementary Network  ...                      NDA   
564     Burnham Park Elementary Network  ...                      NDA   
565       West Side High School Network  ...                       76   

    X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0    1171699.458  1915829.428  41.924497 -87.644522                     7   
1    1196129.985  1856209.466  41.760324 -87.556736                    43   
2

In [35]:
chigago=sqlite3.connect('jupyter_sql.db')

chicago_2=sqlite3.connect('jupyter_sql.db')

In [34]:
chicago_1=sqlite3.connect('jupyter_sql.db')

In [18]:
df2.to_sql('Chicago_Crime',chigago)

ValueError: Table 'Chicago_Crime' already exists.

In [37]:
df3.to_sql('Chigago_Census_Data',chicago_1)


78

In [46]:
%%sql
Select * from Chigago_Census_Data
limit 5

 * sqlite:///jupyter_sql.db
Done.


[(0, 1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (1, 2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (2, 3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (3, 4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (4, 5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)]

In [ ]:
Select 

In [25]:
%%sql
Select COMMUNITY_AREA_NAME from Chicago_Census_Data where PER_CAPITA_INCOME <11000


 * sqlite:///jupyter_sql.db
(sqlite3.OperationalError) no such column: PER_CAPITA_INCOME
[SQL: Select COMMUNITY_AREA_NAME from Chicago_Census_Data where PER_CAPITA_INCOME <11000]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


In [14]:
df4.to_sql('Chicago_Public_School',chicago_2)

566

In [26]:
%%sql
select * from Chicago_Public_School
limit 3

 * sqlite:///jupyter_sql.db
Done.


[(0, 610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl ', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Strong', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', 2.0, '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 1171699.458, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (1, 610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr ', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Strong', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', 15.7, '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (2, 610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave ', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'NDA', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', 2.3, '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)')]

In [7]:

%load_ext sql
%sql sqlite:///jupyter_sql.db


In [19]:
%%sql
SELECT *
 from Chicago_Crime Limit 5

 * sqlite:///jupyter_sql.db
Done.


[(0, 3512276, 'HK587712', '08/28/2004 05:50:56 PM', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, '02/10/2018 03:50:01 PM', 41.8074405, -87.70395585, '(41.8074405, -87.703955849)'),
 (1, 3406613, 'HK456306', '06/26/2004 12:40:00 PM', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, '02/28/2018 03:56:25 PM', 41.89827996, -87.71640551, '(41.898279962, -87.716405505)'),
 (2, 8002131, 'HT233595', '04/04/2011 05:45:00 AM', '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 0, 0, 221, 2, 3.0, 38.0, '6', 1177436.0, 1876313.0, 2011, '02/10/2018 03:50:01 PM', 41.81593313, -87.62464213, '(41.815933131, -87.624642127)'),
 (3, 7903289, 'HT133522', '12/30/2010 04:30:00 PM', '083XX S KINGSTON AVE', '840', 'THEFT', 'FINANCIAL ID THEFT: OVER $300', 'RESIDENCE', 0, 0, 423, 4, 7.0, 46.0, '6', 1194622.0, 1850125.0, 2010, '02/10/2018 03:50:01 PM', 41.74366532, -87.56246276, '(41.743665322, -87.562462756)'),
 (4, 10402076, 'HZ138551', '02/02/2016 07:30:00 PM', '033XX W 66TH ST', '820', 'THEFT', '$500 AND UNDER', 'ALLEY', 0, 0, 831, 8, 15.0, 66.0, '6', 1155240.0, 1860661.0, 2016, '02/10/2018 03:50:01 PM', 41.7734553, -87.70648047, '(41.773455295, -87.706480471)')]

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [13]:
%%sql
SELECT count(*) from Chicago_Crime

 * sqlite:///jupyter_sql.db
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [49]:
%%sql
select COMMUNITY_AREA_NAME from Chigago_Census_Data where PER_CAPITA_INCOME < 11000 


 * sqlite:///jupyter_sql.db
(sqlite3.OperationalError) no such column: PER_CAPITA_INCOME
[SQL: select COMMUNITY_AREA_NAME from Chigago_Census_Data where PER_CAPITA_INCOME < 11000]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [17]:
%%sql 
SELECT CASE_NUMBER from Chicago_Crime 
where description like '%MINOR'

 * sqlite:///jupyter_sql.db
Done.


[('HL266884',), ('HK238408',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [20]:
%%sql 
SELECT * from Chicago_Crime 
where PRIMARY_TYPE like '%KIDNAP%'and DESCRIPTION like '%CHILD%'

 * sqlite:///jupyter_sql.db
Done.


[(520, 5276766, 'HN144152', '01/26/2007 10:05:00 AM', '050XX W VAN BUREN ST', '1792', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER', 'STREET', 0, 0, 1533, 15, 29.0, 25.0, '20', 1143050.0, 1897546.0, 2007, '02/28/2018 03:56:25 PM', 41.87490841, -87.75024931, '(41.874908413, -87.750249307)')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [24]:
%%sql 
SELECT distinct(PRIMARY_TYPE) from Chicago_Crime 
where LOCATION_DESCRIPTION like 'SCHOOL%'

 * sqlite:///jupyter_sql.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the average safety score for each type of school.


In [31]:
%%sql
Select "Elementary, Middle, or High School",  avg(SAFETY_SCORE)from Chicago_Public_School
group by "Elementary, Middle, or High School"


 * sqlite:///jupyter_sql.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [52]:
%%sql
select COMMUNITY_AREA_NAME, PERCENT HOUSEHOLDS BELOW POVERTY from Chicago_Census_Data

 * sqlite:///jupyter_sql.db
(sqlite3.OperationalError) near "BELOW": syntax error
[SQL: select COMMUNITY_AREA_NAME, PERCENT HOUSEHOLDS BELOW POVERTY from Chicago_Census_Data]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


### Problem 8

##### Which community area is most crime prone?


In [8]:
%%sql
select COMMUNITY_AREA_NUMBER, count(*) as frequency from Chicago_Crime
group by COMMUNITY_AREA_NUMBER order by frequency desc limit 1

 * sqlite:///jupyter_sql.db
Done.


[(25.0, 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [43]:
%%sql
select COMMUNITY_AREA_NAME from Chigago_Census_Data where HARDSHIP_INDEX= (select max(HARDSHIP_INDEX) from Chigago_Census_Data)

 * sqlite:///jupyter_sql.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [50]:
%%sql
select COMMUNITY_AREA_NAME from Chigago_Census_Data where COMMUNITY_AREA_NUMBER=(select COMMUNITY_AREA_NUMBER from Chicago_Crime where COMMUNITY_AREA_NUMBER=25 limit 1)

 * sqlite:///jupyter_sql.db
Done.


[('Austin',)]

Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                              |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
